NBitcoin/Protocol/Node.cs

```diff
@@ -144,13 +144,14 @@ public NodeConnection(Node node, Socket socket)
 			{
 				_Node = node;
 				_Socket = socket;
-				_Disconnected = new ManualResetEvent(false);

```

Comment:
Why move this out of the constructor?

Reply:
I moved this out of the constructor because when the NodeConnection class gets created the _Disconnected and _Cancel only gets used when we call BeginListen()

As there are "try" "catches" and possible points of failure, I wanted to instantiate the those objects as late as possible. I.e. I don't want to create them unnecessarily if a possible exception can be thrown before they get used. 

They get set in NodeConnection.BeginListen().

Reply:
But keep it like this don't change back.